apps/storybook/src/hooks/use-upload-attestation.ts

```diff
@@ -0,0 +1,89 @@
+import { getAttestationByUid } from "@/lib/eas/get-attestation-from-uid";
+import {
+	createLighthouseParams,
+	getLighthouseGatewayUrl,
+	uploadEncryptedFileWithText,
+	uploadText,
+} from "@/lib/filecoin/lighthouse/isomorphic";
+import { UploadAttestationParams } from "@/stories/attestations/attestations";
+import { useMutation } from "@tanstack/react-query";
+import { useWalletClient } from "wagmi";
+import { useToast } from "./use-toast";
+
+const LIGHTHOUSE_API_KEY =
+	import.meta.env.VITE_LIGHTHOUSE_API_KEY ||
+	import.meta.env.LIGHTHOUSE_API_KEY! ||
+	import.meta.env.STORYBOOK_LIGHTHOUSE_API_KEY;
+
+export function useUploadAttestation() {
+	const { data: walletClient } = useWalletClient();
+	const { toast } = useToast();
+
+	const mutation = useMutation({
+		mutationFn: async ({
+			uid,
+			payload,
+			chainId = 1,
+			isEncrypted,
+		}: UploadAttestationParams) => {
+			if (!walletClient) return;
+
+			if ((uid || "").length > 0) {
+				const attestation = await getAttestationByUid(uid!, chainId);
+				if (!attestation.attestation) throw new Error("attestation not found");
+				payload = attestation;
+			}
+
+			if (!payload) {
+				console.error(
+					"at least attestation payload or uid with chain must be present",
+				);
+				return;
+			}
+			// default to payload
+			const [apiKey, accountAddress, signedMessage] =
+				await createLighthouseParams({
```

Comment:
i think my existing lighthouse implementation mislead you , with api key we could skip the account

(or we get api key base on account https://docs.lighthouse.storage/lighthouse-1/how-to/create-an-api-key)

I will follow up this one